src/main/java/org/osiam/storage/query/UserFilterParser.java

```diff
@@ -50,4 +50,8 @@ public UserFilterParser(ExtensionDao extensionDao) {
         return UserQueryField.fromString(sortBy.toLowerCase(Locale.ENGLISH));
     }
 
+    @Override
+    protected UserSimpleFilterChain createFilterChain(FilterExpression<UserEntity> filter) {
+        return new UserSimpleFilterChain(entityManager.getCriteriaBuilder(), extensionDao, (UserFilterExpression) filter);
```

Comment:
It works, if you change:

1. [this line](https://github.com/osiam/osiam/pull/108/files#diff-a67f658e1bef24b05cb4e486780572dcR38) to `private final FilterExpression<UserEntity> filterExpression;`

2. [this line](https://github.com/osiam/osiam/pull/108/files#diff-a67f658e1bef24b05cb4e486780572dcR40) to `private final QueryField<UserEntity> userFilterField;`

3. [this line](https://github.com/osiam/osiam/pull/108/files#diff-a67f658e1bef24b05cb4e486780572dcR46) to `FilterExpression<UserEntity> filterExpression`

4. [this line](https://github.com/osiam/osiam/pull/108/files#diff-a67f658e1bef24b05cb4e486780572dcR59) to `private ExtensionQueryField getExtensionFilterField(FilterExpression<UserEntity> filterExpression) {`

You don't need the concrete classes there anymore. All what counts, is that the given instances implement the interfaces, because you've hidden the actual classes used in factories like the sub-classes of `FilterExpression<T>`.